src/yunohost/user.py

```diff
@@ -177,14 +177,17 @@ def user_create(auth, username, firstname, lastname, mail, password,
         try:
             with open('/etc/ssowat/conf.json.persistent') as json_conf:
                 ssowat_conf = json.loads(str(json_conf.read()))
+        except ValueError:
+            raise MoulinetteError(errno.EINVAL,
+                                  m18n.n('ssowat_persistent_conf_error'))
+        except IOError:
+            ssowat_conf = {}
 
-            if 'redirected_urls' in ssowat_conf and '/' in ssowat_conf['redirected_urls']:
-                del ssowat_conf['redirected_urls']['/']
-
-            with open('/etc/ssowat/conf.json.persistent', 'w+') as f:
-                json.dump(ssowat_conf, f, sort_keys=True, indent=4)
+        if 'redirected_urls' in ssowat_conf and '/' in ssowat_conf['redirected_urls']:
+            del ssowat_conf['redirected_urls']['/']
 
-        except IOError: pass
+        with open('/etc/ssowat/conf.json.persistent', 'w+') as f:
+            json.dump(ssowat_conf, f, sort_keys=True, indent=4)
```

Comment:
Maybe I'm wrong, but the `IOError` seems to be handled sooner: https://github.com/YunoHost/yunohost/pull/170/files/937f26bdaaca64849570c472914e8d8a2e7f1fbf#diff-f4b72011f32f548ac5655aec63be4d7fR183 (when the file is open in read-only mode. I don't know if it catch a *no space left* error)